Components/Adyen/Mapper/PaymentMethodMapper.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AdyenPayment\Components\Adyen\Mapper;
+
+use AdyenPayment\Models\Payment\PaymentMethod;
+
+class PaymentMethodMapper implements PaymentMethodMapperInterface
+{
+    public function mapFromAdyen(array $data): \Generator
+    {
+        if ($data['paymentMethods'] ?? []) {
+            foreach ($data['paymentMethods'] as $paymentMethod) {
+                yield PaymentMethod::fromRawPaymentData($paymentMethod);
+            }
+        }
+
+        if ($data['storedPaymentMethods'] ?? []) {
+            foreach ($data['storedPaymentMethods'] as $paymentMethod) {
+                yield PaymentMethod::fromRawPaymentData($paymentMethod);
```

Comment:
@bortefi I know from the description of the PR that store methods are not included, but it seems like this is going to try to save each stored method as a new Shopware method, is that the case?

I guess that multiple customers with multiple stored methods would translate into a large amount of Shopware methods. What we do in Shopware 6 is create 1 Stored Payment Method in Shopware that will list and process payments for all Adyen stored methods, how does that sound?

Reply:
@acampos1916 

Stored payment methods are indeed going to be stored for the users,
this is needed to be able to `remove the current way of overriding shopware 5 everywhere payment means are used.
If they can be treated as actual Shopware payment means, the issue is resolved.

The current way these are imported is not yet optimal, 
and a future improvement ticket will need to address the import process.

Maybe we can discuss the approach of stored payment methods through a call or mail?

Reply:
@bortefi a call sounds great, I'll set it up.